model/otlpgrpc/metrics.go

```diff
@@ -135,7 +136,36 @@ func (mr MetricsRequest) MarshalJSON() ([]byte, error) {
 
 // UnmarshalJSON unmarshalls MetricsRequest from JSON bytes.
 func (mr MetricsRequest) UnmarshalJSON(data []byte) error {
-	return jsonUnmarshaler.Unmarshal(bytes.NewReader(data), mr.orig)
+	if err := jsonUnmarshaler.Unmarshal(bytes.NewReader(data), mr.orig); err != nil {
+		return err
+	}
+	mr.instrumentationLibraryMetricsToScope()
+	return nil
+}
+
+// instrumentationLibraryMetricsToScope implements the translation of resource metrics data
+// following the v0.15.0 upgrade:
+//    	receivers SHOULD check if instrumentation_library_metrics is set
+// 		and scope_metrics is not set then the value in instrumentation_library_metrics
+// 		SHOULD be used instead by converting InstrumentationLibraryMetrics into ScopeMetrics.
```

Comment:
There's a mix of tabs and spaces here, too.

Reply:
Fixed